libs/wire-api/src/Wire/API/Password.hs

```diff
@@ -14,38 +15,84 @@
 --
 -- You should have received a copy of the GNU Affero General Public License along
 -- with this program. If not, see <https://www.gnu.org/licenses/>.
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
 
-module Wire.API.Password
-  ( Password,
-    genPassword,
-    mkSafePassword,
-    verifyPassword,
-  )
-where
+module Wire.API.Password where
+
+-- ( Password,
+--   genPassword,
+--   mkSafePassword,
+--   verifyPassword,
+-- )
 
 import Cassandra
-import Crypto.Scrypt
+import Crypto.KDF.Scrypt as Scrypt
+import Crypto.Random
+import Data.ByteArray hiding (length)
 import Data.ByteString.Base64 qualified as B64
+import Data.ByteString.Char8 qualified as C8
 import Data.ByteString.Lazy (fromStrict, toStrict)
 import Data.Misc
+import Data.Text qualified as Text
 import Data.Text.Encoding qualified as Text
 import Imports
 import OpenSSL.Random (randBytes)
 
 -- | A derived, stretched password that can be safely stored.
 newtype Password = Password
-  {fromPassword :: EncryptedPass}
+  {fromPassword :: Text}
 
 instance Show Password where
   show _ = "<Password>"
 
 instance Cql Password where
   ctype = Tagged BlobColumn
 
-  fromCql (CqlBlob lbs) = pure . Password . EncryptedPass $ toStrict lbs
+  fromCql (CqlBlob lbs) = pure . Password . Text.decodeUtf8 . toStrict $ lbs
   fromCql _ = Left "password: expected blob"
 
-  toCql = CqlBlob . fromStrict . getEncryptedPass . fromPassword
+  toCql = CqlBlob . fromStrict . Text.encodeUtf8 . fromPassword
```

Comment:
How does this interact with existing data in the DB?

Reply:
It is still the same algorithm, it should be 100% the same 🙏

Reply:
Do we have any golden and roundtrip tests for this? They'd make me feel easier about this change :)

Reply:
I think I can property test it 👍 let's see